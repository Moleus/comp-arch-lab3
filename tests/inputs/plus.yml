translator_input: |-
  a: word: 5
  b: word: 10
  start: ld a
    add b
  hlt

machine_input: |-
  [{ "arrivesAt": 0, "char": "a" }]
