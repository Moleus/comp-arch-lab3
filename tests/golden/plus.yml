translator_output: |-
    {
      "StartAddress": 2,
      "Instructions": [
        {
          "index": 0,
          "label": "a",
          "opcode": "NOP",
          "operand": 5,
          "term_info": {
            "line_num": 1,
            "original_content": "a: word: 5"
          }
        },
        {
          "index": 1,
          "label": "b",
          "opcode": "NOP",
          "operand": 10,
          "term_info": {
            "line_num": 2,
            "original_content": "b: word: 10"
          }
        },
        {
          "index": 2,
          "label": "start",
          "opcode": "LD",
          "operand": 0,
          "term_info": {
            "line_num": 3,
            "original_content": "start: ld a"
          }
        },
        {
          "index": 3,
          "opcode": "ADD",
          "operand": 1,
          "term_info": {
            "line_num": 4,
            "original_content": "add b"
          }
        },
        {
          "index": 4,
          "opcode": "HLT",
          "term_info": {
            "line_num": 5,
            "original_content": "hlt"
          }
        }
      ]
    }
stdout: '[{ "arrivesAt": 0, "char": "a" }]'
log: |
    IP -> AR                      | DR:  0, AR:  2, AC:  0, IP:  2, CR:  0, PS:  0, SP:  0 | !Z !N !C | mem[AR]: {10 0 0} | CR: NOP 0
    IP + 1 -> IP; mem[AR] -> DR   | AC:  0, IP:  3, CR:  0, PS:  0, SP:  0, DR:  0, AR:  2 | !Z !N !C | mem[AR]: {10 0 0} | CR: NOP 0
    DR -> CR                      | IP:  3, CR:  0, PS:  0, SP:  0, DR:  0, AR:  2, AC:  0 | !Z !N !C | mem[AR]: {10 0 0} | CR: LD
    DR -> AR                      | SP:  0, DR:  0, AR:  0, AC:  0, IP:  3, CR:  0, PS:  0 | !Z !N !C | mem[AR]: {0 5 0} | CR: LD
    mem[AR] -> DR                 | DR:  5, AR:  0, AC:  0, IP:  3, CR:  0, PS:  0, SP:  0 | !Z !N !C | mem[AR]: {0 5 0} | CR: LD
    DR -> AC                      | IP:  3, CR:  0, PS:  0, SP:  0, DR:  5, AR:  0, AC:  5 | !Z !N !C | mem[AR]: {0 5 0} | CR: LD

    IP -> AR                      | AC:  5, IP:  3, CR:  0, PS:  0, SP:  0, DR:  5, AR:  3 | !Z !N !C | mem[AR]: {3 1 0} | CR: LD
    IP + 1 -> IP; mem[AR] -> DR   | AR:  3, AC:  5, IP:  4, CR:  0, PS:  0, SP:  0, DR:  1 | !Z !N !C | mem[AR]: {3 1 0} | CR: LD
    DR -> CR                      | SP:  0, DR:  1, AR:  3, AC:  5, IP:  4, CR:  1, PS:  0 | !Z !N !C | mem[AR]: {3 1 0} | CR: ADD
    DR -> AR                      | SP:  0, DR:  1, AR:  1, AC:  5, IP:  4, CR:  1, PS:  0 | !Z !N !C | mem[AR]: {0 10 0} | CR: ADD
    mem[AR] -> DR                 | IP:  4, CR:  1, PS:  0, SP:  0, DR: 10, AR:  1, AC:  5 | !Z !N !C | mem[AR]: {0 10 0} | CR: ADD
    AC +- DR -> AC                | AC: 15, IP:  4, CR:  1, PS:  0, SP:  0, DR: 10, AR:  1 | !Z !N !C | mem[AR]: {0 10 0} | CR: ADD

    IP -> AR                      | AC: 15, IP:  4, CR:  1, PS:  0, SP:  0, DR: 10, AR:  4 | !Z !N !C | mem[AR]: {6 -1 0} | CR: ADD
    IP + 1 -> IP; mem[AR] -> DR   | SP:  0, DR: -1, AR:  4, AC: 15, IP:  5, CR:  1, PS:  0 | !Z !N !C | mem[AR]: {6 -1 0} | CR: ADD
    DR -> CR                      | CR: -1, PS:  0, SP:  0, DR: -1, AR:  4, AC: 15, IP:  5 | !Z !N !C | mem[AR]: {6 -1 0} | CR: HLT
    HLT                           | AC: 15, IP:  5, CR: -1, PS:  0, SP:  0, DR: -1, AR:  4 | !Z !N !C | mem[AR]: {6 -1 0} | CR: HLT
